### PR TITLE
[Doc] Update doc comments for IJSValueReader and IJSValueWriter

### PR DESCRIPTION
**Pull request by vmoroz**
This PR contains doc comment changes for `IJSValueReader` and `IJSValueWriter`.
There are no functional changes.
The corresponding docs repo PR: https://github.com/microsoft/react-native-windows-samples/pull/357

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6891)

### CODE REVIEW

**Review by NikoAri — APPROVED** [review]
:shipit:

File: vnext/Microsoft.ReactNative/IJSValueReader.idl [hunk 1]
```diff
@@ -3,10 +3,11 @@
 
 #include "DocString.h"
 
-namespace Microsoft.ReactNative {
-
-  // Type of value read from JavaScript by IJSValueReader
-  enum JSValueType {
+namespace Microsoft.ReactNative
+{
+  DOC_STRING("Type of value read by IJSValueReader.")
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/IJSValueReader.idl:8]: moving the comments from the markdown PR to here :)
```suggestion
  DOC_STRING("Type of value read by @IJSValueReader.")
```

File: vnext/Microsoft.ReactNative/IJSValueReader.idl [hunk 2]
```diff
@@ -16,29 +17,69 @@ namespace Microsoft.ReactNative {
...
+    "When `IJSValueReader` reads data it must walk the whole tree without skipping any items. "
+    "It means that if the current value type is `Object`, then we must call @.GetNextObjectProperty "
+    "to start reading its properties, and if the current type is `Array`, then we must call "
+    "@.GetNextArrayItem to start reading its items.\n"
+    "\n"
+    "See the @IJSValueWriter for the corresponding writer interface.\n"
+    "\n"
+    "The @IJSValueReader and @IJSValueWriter should be rarely used directly. "
+    "They are usually used to create serializer and deserializer functions for a type. "
+    "In the rest of our code we must use these functions to serialize/deserialize our values. "
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/IJSValueReader.idl:39]: whose code does "our code" refer to? Can we replace it with something specific, e.g. either "app code", "React Native for Windows framework code", etc?

File: vnext/Microsoft.ReactNative/IJSValueReader.idl (hunk 1 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/IJSValueReader.idl]: Updated. Thank you!

File: vnext/Microsoft.ReactNative/IJSValueReader.idl (hunk 2 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/IJSValueReader.idl]: I have changed the wording.

**Review by asklar — APPROVED** [review]

**Review by jonthysell — APPROVED** [review]

**Pull request merged**